### PR TITLE
Move LLVM dependency to an external repository

### PR DESCRIPTION
**Pull request by shahms**
The generated_cmake_targets.bzl file was generated by a separate tool(s) which I'm working on getting in shape to push to a separate repository, but for now this should get the LLVM/Clang build working as an external repo, obviating and removing the `tools/modules/` scripts.

I'm currently running the tests on Darwin, but so far things look relatively good.

### CODE REVIEW

File: tools/build_rules/llvm/cmakedefines/BUILD [hunk 1]
```diff
@@ -0,0 +1,14 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["cmakedefines.go"],
+    importpath = "github.com/shahms/cmaketobzl/tools/cmakedefines",
```
Comment (jaysachs) [inline tools/build_rules/llvm/cmakedefines/BUILD]: Is that the canonical location?

Comment (shahms) [reply tools/build_rules/llvm/cmakedefines/BUILD]: That was where it was during development, but the current location is canonical for now (although fairly academic as nothing should reference it by that name).

File: tools/build_rules/llvm/cmakedefines/cmakedefines.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+	case bool:
+		*cv = cmakeValue(strings.ToUpper(fmt.Sprint(v)))
+	case float64:
+		*cv = cmakeValue(fmt.Sprint(v))
+	case string:
+		*cv = cmakeValue(v)
+	case nil:
+		*cv = ""
+	default:
+		return fmt.Errorf("expected bool, string or number, found: %v", data)
```
Comment (schroederc) [inline tools/build_rules/llvm/cmakedefines/cmakedefines.go]: You can use `%T` to print the type.

Comment (shahms) [reply tools/build_rules/llvm/cmakedefines/cmakedefines.go]: Done.

**Pull request merged**